assets/javascripts/discourse/components/category-tag-chat-link.js

```diff
@@ -0,0 +1,52 @@
+import Component from "@ember/component";
+import { ajax } from "discourse/lib/ajax";
+import { action } from "@ember/object";
+import { schedule } from "@ember/runloop";
+
+export default Component.extend({
+  category: null,
+  tag: null,
+  show: false,
+
+  init() {
+    this._super(...arguments);
+
+    if (!this.currentUser?.has_chat_enabled) {
+      return;
+    }
+
+    if (this.category?.custom_fields.has_chat_enabled) {
```

Comment:
Is it possible `custom_fields` will be null?

Reply:
It's possible for `this.category` to be null, but not `custom_fields` as long as I know. Guess I could add another `?` if that helps

Reply:
Nah just leave it